komga-webui/src/components/dialogs/EditSeriesDialog.vue

```diff
@@ -624,6 +694,61 @@ export default Vue.extend({
         return true
       } else return false
     },
+    inputChangeHandler: function (files: File[]) {
+      for (const file of files) {
+        if (!this.uploadQueue.includes(file)) {
+          this.uploadQueue.push(file)
+        }
+      }
+    },
+    getThumbnails: async function () {
+      if (Array.isArray(this.series)) return
+
+      const thumbnails = await this.$komgaSeries.getThumbnails(this.series.id)
+
+      this.selectThumbnailHandler(thumbnails.find(({selected}) => selected))
+
+      this.seriesThumbnails = thumbnails
+    },
+    getKey: function (item: File | SeriesThumbnailDto) {
+      return item instanceof File ? item.name : item.id
+    },
+    isSelected: function (item: File | SeriesThumbnailDto): boolean {
+      return item instanceof File ? item.name === this.selectedThumbnail : item.id === this.selectedThumbnail
+    },
+    selectThumbnailHandler: function (item: File | SeriesThumbnailDto | undefined) {
+      if (!item) {
+        return
+      } else if (item instanceof File) {
+        this.selectedThumbnail = item.name
+      } else {
+        this.selectedThumbnail = item.id
+      }
+    },
+    toBeDeleted: function (item: File | SeriesThumbnailDto) {
+      if (item instanceof File) {
+        return false
+      } else {
+        return this.deleteQueue.includes(item)
+      }
+    },
+    deleteThumbnailHandler: function (item: File | SeriesThumbnailDto) {
+      if (item instanceof File) {
+        const index = this.uploadQueue.indexOf(item, 0)
+        if (index > -1) {
+          this.uploadQueue.splice(index, 1)
+        }
+      } else {
+        if (this.toBeDeleted(item)) {
+          const index = this.deleteQueue.indexOf(item, 0)
+          if (index > -1) {
+            this.deleteQueue.splice(index, 1)
+          }
```

Comment:
why replace the item if it is already in the array?

Reply:
To make delete togglable that way one doesn't need to reset if mistakenly pressing delete.

Why uploads are just removed is because they are not uploaded and can just be added again via select or drag & drpå